qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -109,170 +108,123 @@ class Grover:
-                                                               mct_mode)
-
-        max_iterations = np.ceil(2 ** (len(self._grover_operator.reflection_qubits) / 2))
-        if incremental:  # TODO remove 3 months after 0.8.0
-            if rotation_counts is not None:
-                iterations = rotation_counts
-                self._sample_from_iterations = False
-            else:
-                if lam is None:
-                    lam = 1.34
-
-                iterations = []
-                self._sample_from_iterations = True
-                power = 1.0
-                while power < max_iterations:
-                    iterations.append(int(power))
-                    power = lam * power
-
-        elif num_iterations is not None:  # TODO remove 3 months after 0.8.0
-            iterations = [num_iterations]
-        elif not isinstance(iterations, list):
-            iterations = [iterations]
-        # else: already a list
-
-        # cutoff if max_iterations is exceeded (legacy code, should considered for removal?)
-        self._iterations = []
-        for iteration in iterations:
-            self._iterations += [iteration]
-            if iteration > max_iterations:
-                break
+            self._iterations = iterations
 
-        # check the type of good_state
-        _check_is_good_state(good_state)
+        self._quantum_instance = None
+        if quantum_instance is not None:
+            self.quantum_instance = quantum_instance
 
-        self._is_good_state = good_state
         self._sample_from_iterations = sample_from_iterations
-        self._post_processing = post_processing
-        self._incremental = incremental
-        self._lam = lam
-        self._rotation_counts = rotation_counts
-
-        if incremental or (isinstance(iterations, list) and len(iterations) > 1):
-            logger.debug('Incremental mode specified, \
-                ignoring "num_iterations" and "num_solutions".')
-
-        self._ret = GroverResult()
 
     @property
     def quantum_instance(self) -> Optional[QuantumInstance]:
-        """ Returns quantum instance. """
+        """Get the quantum instance.
+        Returns:
+            The quantum instance used to run this algorithm.
+        """
         return self._quantum_instance
 
     @quantum_instance.setter
     def quantum_instance(self, quantum_instance: Union[QuantumInstance,
                                                        BaseBackend, Backend]) -> None:
-        """ Sets quantum instance. """
+        """Set quantum instance.
+        Args:
+            quantum_instance: The quantum instance used to run this algorithm.
+        """
         if isinstance(quantum_instance, (BaseBackend, Backend)):
             quantum_instance = QuantumInstance(quantum_instance)
         self._quantum_instance = quantum_instance
 
+    def amplify(self, amplification_problem: AmplificationProblem) -> 'GroverResult':
```

Comment:
Shouldn't this method signature be in an interface for an AmplitudeAmplifier and Grover implements this as per here

Reply:
yes, I added the interface class in 8a6e178